.github/workflows/changed_files.yml

```
@@ -0,0 +1,133 @@
+name: List files changed as GitHub comment
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  list-files-changed:
+    if: github.event.pull_request.head.repo.full_name == github.repository
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+
+      - name: Format list of changed files
+        id: format
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          export output=()
+          export all_changed_files=()
+          export added_files=()
+          export modified_files=()
+          export removed_files=()
+          export no_preview_files=()
+
+          export pr_num="${{ github.event.pull_request.number }}"
+
+          export all_changed_files="$(gh api repos/wandb/docs/pulls/${pr_num}/files -q '.[] | [.status,.filename] | @tsv' | sort)"
+
+          for changed_file in "${all_changed_files[@]}"; do
+            export status=''
+            export path=''
+            echo "$changed_file" | while read i; do
+              status="$(echo $i | awk -F ' ' {'print $1'})"
+              echo "Status: $status"
+              path="$(echo $i | awk -F ' ' {'print $2'})"
+              echo "Path: $path"
+              case "$status" in
+                "added")
+                  added_files+=("$path")
+                  echo "${added_files[@]}"
+                  ;;
+                "modified")
+                  modified_files+=("$path")
+                  echo "${modified_files[@]}"
+                  ;;
+                "removed")
+                  removed_files+=("$path")
+                  echo "${removed_files[@]}"
+                *)
+                  echo "Unknown status $status for file $path. Giving up."
+                  exit -1
+                  ;;
+              esac
+            done
+          done
+
+          echo "Added:"
+          echo "${added_files[@]}"
+          echo
+
+          echo "Modified:"
+          echo "${modified_files[@]}"
+          echo
+
+          echo "Removed:"
+          echo "${removed_files[@]}"
+          echo
+
+          function generatePreviewURL() {
```

Combining () and {} might be causing errors. i.e.

```
function_name () {
  commands
}
```

or 
```
function function_name {
   commands
}
```
are OK

```suggestion
          function generatePreviewURL {
```